src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -59,6 +59,22 @@ public static bool IsBlittableType(TypeDesc type)
             return false;
         }
 
+        public static bool IsStructMarshallingRequired(TypeDesc typeDesc)
+        {
+            if (typeDesc.Category != TypeFlags.ValueType)
+                return false;
+
+
+            var type = typeDesc.UnderlyingType as MetadataType;
```

Comment:
This could be a ByRefType as well. I will keep the signature as TypeDesc, add handling of ByRefType and return false if the type is not a MetadataType.